dev-libs/crypto++/crypto++-5.6.5.recipe

```diff
@@ -48,24 +53,38 @@ BUILD_PREREQUIRES="
 	cmd:gcc$secondaryArchSuffix
 	cmd:ld
 	cmd:make
-	cmd:pkg_config$secondaryArchSuffix
 	cmd:nasm
+	cmd:pkg_config$secondaryArchSuffix
 	"
 
+defineDebugInfoPackage crypto++$secondaryArchSuffix \
+	$libDir/libcryptopp.so.$libVersion
+
 BUILD()
 {
-	make shared dynamic static $jobArgs
+	make HAS_SOLIB_VERSION=1 dynamic static $jobArgs
 }
 
 INSTALL()
 {
-	make install PREFIX=$prefix INCLUDEDIR=$includeDir
-
-	mv $libDir/libcryptopp.so $libDir/libcryptopp.so.$portVersion
-	ln -s $libDir/libcryptopp.so.$portVersion $libDir/libcryptopp.so
+	make \
+		HAS_SOLIB_VERSION=1 LDCONF=true \
+		BINDIR=$binDir LIBDIR=$libDir \
+		DATADIR=$dataDir INCLUDEDIR=$includeDir \
+		install
 
+	# Yes, they created the link. But they forgot to install it...
+	ln -s libcryptopp.so.$libVersion $libDir/libcryptopp.so.$libVersionCompat
 	prepareInstalledDevelLib libcryptopp
 
+	install -d -m 755 $developLibDir/pkgconfig
+	cat > $developLibDir/pkgconfig/libcrypto++.pc << EOF
```

Comment:
When I try to build codecrypt (after having installed crypto++ with your changes), I get:
```
checking for pkg-config... /bin/pkg-config
checking pkg-config is at least version 0.25... yes
checking for CRYPTOPP... no
configure: error: Package requirements (libcrypto++) were not met:

No package 'libcrypto++' found
```
The libcrypto++.pc we get has this:
```
Name: libcrypto++-5.6.5
Description: Class library of cryptographic schemes
Version: 5.6.5
Libs: -lcryptopp
```
But I think it should look like this:
```
prefix=/packages/crypto++-5.6.5-3/.self
exec_prefix=${prefix}
libdir=${prefix}/develop/lib
includedir=${prefix}/develop/headers
Name: libcrypto++-5.6.5
Description: Class library of cryptographic schemes
Version: 5.6.5
Libs: -L${libdir} -lcryptopp
```
I liked your idea of having a template in additional-files. You could put:
```
prefix=@PREFIX@
exec_prefix=${prefix}
libdir=${prefix}/develop/lib
includedir=${prefix}/develop/headers
Name: libcrypto++-@VERSION@
Description: Class library of cryptographic schemes
Version: @VERSION@
Libs: -L${libdir} -lcryptopp
```
in libcrypto++.pc.in and then use
```
	sed -i \
		-e "s|@PREFIX@|$prefix|;" \
		-e "s|@VERSION@|$portVersion|;" \
		$portDir/additional-files/libcrypto++.pc.in \
		> $developLibDir/pkgconfig/libcrypto++.pc
```

Reply:
Regarding the external file, @extrowerk [suggested](https://github.com/haikuports/haikuports/pull/1900#discussion_r157337733) against it.

Are you sure you're using my updated crypto++? It's working well here on both x86_gcc2 and x86_64.

And no, the only reason .pc looks bloated like so is for autotools interop. Since our library is in the compiler search path, it's redundant to put lib path in the .pc file

Reply:
You are right. I think that, as you suspected, I had the old crypto++ when I tried to build codecrypt the first time.
(So sorry for the false alert.)

Hmm, all the other pc files I have on my system don't have the version appended to the Name: field. So maybe you can make this tiny change:
```diff
-Name: libcrypto++-$portVersion
+Name: libcrypto++
```